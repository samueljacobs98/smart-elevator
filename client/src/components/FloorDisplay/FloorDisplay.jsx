import "./FloorDisplay.scss";

const FloorDisplay = ({ floor }) => {
  return (
    <h1 className="floor-display">
      You are on <span className="floor-display__floor">Floor {floor}</span>
    </h1>
  );
};

export default FloorDisplay;
