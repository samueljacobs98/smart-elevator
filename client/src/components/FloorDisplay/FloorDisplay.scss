@use "../../assets/scss/variables" as *;

.floor-display {
  font-size: $font-size-large;
  font-weight: $font-weight-light;
  text-align: center;
  padding: 1rem;
  border-bottom: 2px solid $colour-grey-dark;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  height: fit-content;

  &__floor {
    font-size: inherit;
    font-weight: $font-weight-bold;
  }

  @media (min-width: $breakpoint-tablet) {
    font-size: $font-size-extra-large;
  }

  @media (min-width: $breakpoint-desktop) {
    font-size: $font-size-extra-extra-large;
  }
}
