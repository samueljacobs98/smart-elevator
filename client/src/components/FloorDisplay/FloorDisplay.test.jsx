import { render, screen } from "@testing-library/react";
import FloorDisplay from "./FloorDisplay";
import "@testing-library/jest-dom";

describe("FloorDisplay", () => {
  const renderComponent = (floor = 1) => render(<FloorDisplay floor={floor} />);

  it("renders without crashing", () => {
    renderComponent();

    expect(screen).toBeDefined();
  });

  it("render correctly with the correct floor number", () => {
    const floorNumber = 5;

    renderComponent(floorNumber);

    const floorDisplay = screen.getByRole("heading");
    const floorDisplayNumber = screen.getByText(`Floor ${floorNumber}`);

    expect(floorDisplay).toBeInTheDocument();
    expect(floorDisplayNumber).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
