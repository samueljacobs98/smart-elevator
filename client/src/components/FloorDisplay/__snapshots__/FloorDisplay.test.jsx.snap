// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorDisplay should match snapshot 1`] = `
<div>
  <h1
    class="floor-display"
  >
    You are on 
    <span
      class="floor-display__floor"
    >
      Floor 
      5
    </span>
  </h1>
</div>
`;
